necroHeal.lic

```diff
@@ -0,0 +1,171 @@
+        hunting_info << info
+      else
+        hunting_info.last['args'] << info['args']
+        hunting_info.last[:duration] << info[:duration]
+        hunting_info.last['stop_on'] << info['stop_on']
+      end
+    end
+    hunting_info
+  end
+
+  ## Copy from Hunting Buddy
+  def find_hunting_room?(zone_name)
+    UserVars.friends = @settings.hunting_buddies || []
+    if rooms = @hunting_zones[zone_name]
+      @exit = nil
+      find_empty_room(rooms, @settings.safe_room,
+                      lambda do
+                        return true if (DRRoom.pcs & UserVars.friends).any?
+                        return false if (DRRoom.pcs - DRRoom.group_members).any?
+                        # No visible friends in the room and no visible people
+                        UserVars.friends.each { |friend| Flags.add("room-check-#{friend}", friend) }
+                        Flags.add('room-check', 'says, ', 'say, ', 'You hear')
+                        fput("say #{@settings.empty_hunting_room_messages.sample}")
+                        bput('search', 'roundtime')
+                        data = reget(40).reverse.take_while { |x| x !~ /You search around/ }
+                        if data.grep(/vague silhouette|You notice \w+, who is|see signs that/).any?
+                          pause
+                          waitrt?
+                          return UserVars.friends.find { |friend| Flags["room-check-#{friend}"] }
+                        end
+                        20.times do |_|
+                          pause 0.5
+                          return true if UserVars.friends.find { |friend| Flags["room-check-#{friend}"] }
+                          return false if Flags['room-check'] || !(DRRoom.pcs - DRRoom.group_members - UserVars.friends).empty?
+                        end
+                        true
+                      end)
+    else
+      escort_info = @escort_zones[zone_name]
+      unless escort_info
+        echo "FAILED TO FIND HUNTING ZONE #{zone_name} IN BASE.YAML"
+        return false
+      end
+      walk_to(escort_info['base'])
+      wait_for_script_to_complete('bescort', [escort_info['area'], escort_info['enter']])
+      @exit = [escort_info['area'], 'exit']
+    end
+
+    true
+  end
+
+  ## new method for running a script and waiting for it to finish
+  def runWaitScript(script, var1)
+    start_script "#{script}", ["#{var1}"] if var1 !~ /xx/
+    start_script "#{script}" if var1 =~ /xx/
+    sleep 1
+    wait_while { running?("#{script}")}
+    waitrt?
+    sleep 1
+  end
+
+  ## Start new code for necrohealing
+  def main
+
+    ## Sets flag to check for game line to determine when no wounds exist
+  
+    ## Flag uses consume logic from combat-training. If no wounds are left
+    ## then combat-training uses dissect or harvest. Regex tests for both rituals.
+    Flags.add('noWounds', '(You carefully position the corpse and make a cut|You bend over .* to make a few quick)')
+    Flags['noWounds'] = false

```

Comment:
I was initially using combat.trainers check_health but didn't want to replicate the the entire game_state variable.  Is there a way to pull status via combat.trainer?

Right now combat-trainer is doing the actual health command so that's why i was using the flag to check. Is there a way for common-health to store health data without actually doing the health command?